packages/ember-data/lib/system/relationships/belongs-to.js

```diff
@@ -86,9 +91,16 @@ function belongsTo(type, options) {
       } else {
         this._relationships[key].setRecord(value);
       }
+
+      return this._relationships[key].getRecord();
     }
+  };
 
-    return this._relationships[key].getRecord();
+  return Ember.computed(supportsComputedGetterSetter ? computedConfig : function(key, value) {
+    if (arguments.length > 1) {
+      return computedConfig.set.call(this, key, value);
+    }
+    return computedConfig.get.call(this, key);
```

Comment:
If we are already putting `supportsComputedGetterSetter` into a shared util, but it's only used as part of this function, we should probably move this into the util as well.